src/common/tusb_fifo.c

```diff
@@ -325,7 +325,7 @@ static uint16_t advance_pointer(tu_fifo_t* f, uint16_t p, uint16_t offset)
   // We are exploiting the wrap around to the correct index
 
   // TODO warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
```

Comment:
The GCC seem to be right about this. I got the warning when compiling with gnu++17, haven't got time to trace it down. So I put the warning comment here.  Thanks for the PR.